include/ed/update_request.h

```diff
@@ -51,6 +51,30 @@ class UpdateRequest
     void setShape(const UUID& id, const geo::ShapeConstPtr& shape) { shapes[id] = shape; flagUpdated(id); }
 
 
+    //VOLUMES
+
+    std::map<UUID, std::map<std::string, geo::ShapeConstPtr> > volumes_added;
+    void addVolume(const UUID& id, const std::string Volume_name, const geo::ShapeConstPtr& Volume_shape)
+    {   std::map<UUID, std::map<std::string, geo::ShapeConstPtr> >::iterator it = volumes_added.find(id);
+        if (it != volumes_added.end())
+        {
+            std::map<std::string, geo::ShapeConstPtr>::iterator it2 = it->second.find(Volume_name);
+            if (it2 != it->second.end())
+                it2->second = Volume_shape;
+            else
+                it->second[Volume_name] = Volume_shape;
+        }
+        else
+        {
+            std::map<std::string, geo::ShapeConstPtr> volume_map;
+            volume_map[Volume_name] = Volume_shape;
+            volumes_added[id] = volume_map;
+        }
+        flagUpdated(id);
+    }
+
+    std::map<UUID, std::set<std::string> > volumes_removed;
```

Comment:
Same for this variable

Reply:
No underscore used in update_request at all. So out of this scope.